### ISSUE TITLE
Incredibly slow to compress, barely useable

### ISSUE THREAD

**Issue by TimvanScherpenzeel** [ralscha/parcel-plugin-compress#6]
I've been looking into pre-compressing my assets with Parcel and I came across this plugin. It has been working fine for the first setup but once I start loading slightly larger assets (couple of hundred KB - 2MB) the compression time explodes. Is this a known issue?

15 iterations
<img width="564" alt="Screen Shot 2019-03-26 at 18 59 51" src="https://user-images.githubusercontent.com/11870513/55021718-8f3b7400-4ff9-11e9-9d3b-364bbb6a728a.png">

1 iteration reduces the compression time but still far too long
<img width="571" alt="Screen Shot 2019-03-26 at 19 00 56" src="https://user-images.githubusercontent.com/11870513/55021726-92366480-4ff9-11e9-8a84-eb5e9b4028d6.png">

**Comment by ralscha** [comment]
The plugin uses [@gfx/zopfli](https://github.com/gfx/universal-zopfli-js) for the gz compression. 
According to the project page:

>This library is a JavaScript binding to zopfli with WebAssembly. This might be slower than C/C++ extension for zopfli, but because wasm is portable binary its installation is much easier than C/C++ extensions.

 What we could do and what we [already do for brotli](https://github.com/ralscha/parcel-plugin-compress/blob/master/brotliAdapter.js) is checking if the C implementation is installed and use that instead:

https://www.npmjs.com/package/node-zopfli-es

Another issue is that zopfli is much slower than gzip
>It is considerably slower than gzip (~100x) so you may want to use it only for static content and cached resources.

Maybe we could add a switch to the plugin where the user can choose between zopfli and gzip.

**Comment by TimvanScherpenzeel** [comment]
Thank you for your quick response :+1:

>What we could do and what we already do for brotli is checking if the C implementation is installed and use that instead: https://www.npmjs.com/package/node-zopfli-es

Would it be an option to use the native `zlib` library built into Node.js? 

https://github.com/Rebsos/node-gzip/blob/master/src/index.js

It also mentions support for `Brotli` compression:
> The zlib module provides compression functionality implemented using Gzip and Deflate/Inflate, as well as Brotli.

https://nodejs.org/api/zlib.html

>Maybe we could add a switch to the plugin where the user can choose between zopfli and gzip.

That would be great. I understand the need for thorough compression but I have found that a fast single pass gzip works well enough in most of my use cases.

**Comment by ralscha** [comment]
Great to see brotli support in zlib. But I guess we should wait switching to zlib for brolit compression until it is in the next node LTS release. 

But for gzip we should add a switch that uses zopfli by default and zlib if you configure it.

**Comment by ralscha** [comment]
## @gfx/zopfli
```
dist\src.dc4d2d1d.js.map.gz    18.26 KB    0.991s
dist\src.dc4d2d1d.js.gz         9.34 KB    0.244s
dist\index.html.gz                159 B    0.095s
```

## node-zopfli-es
```
dist\src.dc4d2d1d.js.map.gz    18.26 KB    0.607s
dist\src.dc4d2d1d.js.gz         9.34 KB    0.136s
dist\index.html.gz                159 B    0.056s
```

**Comment by ralscha** [comment]
I released a new version 1.0.8

It uses by default `node-zopfli-es` for gzip compression.
You can enable zlib with the following configuration. 

`package.json`
```
"compress": {
   "gzip": {   
      "zlib": true
  },
}
```

or `.compressrc`
```
{
  "gzip": {
    "zlib": true
  }
}
```

**Comment by TimvanScherpenzeel** [comment]
Thank you! 👍
